File: baekjoon/단계 별로 풀어보기/1차원 배열/4344_평균은_넘겠지.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main()
{
    cin.tie(NULL);
	cout.tie(NULL);
	ios_base::sync_with_stdio(false);
	cout << fixed;
	cout.precision(3);
	
	int T, N, sum, counter;
	float average;
	cin >> T;
	
	
	
	for (int i = 0; i < T; i++)
	{
	    sum = 0;
	    counter = 0;
	    
	    cin >> N;
	    int *data = new int[N];
	    
	    for (int j = 0; j < N; j++)
	    {
	        cin >> data[j];
	        sum += data[j];
	    }
	    
	    average = (float)(sum) / N;
	    
	    for (int j = 0; j < N; j++)
	    {
	        if (average < data[j]) counter++;
	    }
	    
	    cout << (float)(counter) / N * 100 << '%' << '\n';
	}
	
}

File: baekjoon/단계 별로 풀어보기/1차원 배열/4344_평균은_넘겠지.py
import sys
input = sys.stdin.readline
IP = lambda: map(int, sys.stdin.readline().split())

def solve():
    for T in range(int(input())):
        N, *data = IP()
        average = sum(data) / N
        counter = 0
        for each in data:
            if average < each: counter += 1
        
        print("%.3lf" % round(counter / N * 100, 3) + "%")
    
if "__main__" == __name__: solve()